plugin/evm/vm.go

```diff
@@ -1019,7 +1020,18 @@ func (vm *VM) CreateHandlers(context.Context) (map[string]http.Handler, error) {
 	}
 
 	if vm.config.WarpAPIEnabled {
-		validatorsState := warpValidators.NewState(vm.ctx)
+		requirePrimaryNetworkSigners := func() bool {
+			warpCfgIntf, ok := vm.currentRules().ActivePrecompiles[warpcontract.ContractAddress]
+			if !ok {
+				return false
+			}
+			warpCfg, ok := warpCfgIntf.(*warpcontract.Config)
+			if !ok {
+				return false
+			}
+			return warpCfg.RequirePrimaryNetworkSigners
+		}
+		validatorsState := warpValidators.NewState(vm.ctx, requirePrimaryNetworkSigners)
```

Comment:
nit: does it make sense to have an interface rather than passing the function (so vm/warpConfig can implement it)?